pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -121,6 +121,15 @@ protected TransferableBlock getNextBlock() {
     }
   }
 
+  @Override
```

Comment:
unable to comment to the code above but my question is on the logic inside the try/catch block for getNextBlock()
```
      if (!_isHashTableBuilt) {
        // Build JOIN hash table
        buildBroadcastHashTable();  // [RR] if this executes successfully after consuming all the right-side blocks; it doesn't return, which will fall through
      }
      if (_upstreamErrorBlock != null) {
        return _upstreamErrorBlock;
      } else if (!_isHashTableBuilt) { // [RR] and hashtable is built here
        return TransferableBlockUtils.getNoOpTransferableBlock();
      }
      // JOIN each left block with the right block. 
      // [RR] this will be trigger even if the original opChain is scheduled by the RIGHT mailbox, correct? b/c after hashtable built, it will continue to consume left block until left EOS, correct?
      return buildJoinedDataBlock(_leftTableOperator.nextBlock());
```

Reply:
that's a good point 🤔 I'm not sure then that this explains the problem... there might be something else that is causing it to miss the notification